### ISSUE TITLE
compiling error

### ISSUE THREAD

**Issue by pvdw0310** [moononournation/MiniTV#2]
Hi, I 'm trying to compile MJPEG_2task_Audio_1task.ino and I get a compiling error
I downloaded all the libraries and get:

In file included from C:\Users\peter\Documents\Arduino\MiniTV\MJPEG_2task_Audio_1task\MJPEG_2task_Audio_1task.ino:40:
esp32_audio_task.h:85:59: error: invalid conversion from 'void (*)(AACFrameInfo&, int16_t*, size_t)' {aka 'void (*)(_AACFrameInfo&, short int*, unsigned int)'} to 'libhelix::AACDataCallback' {aka 'void (*)(_AACFrameInfo&, short int*, unsigned int, void*)'} [-fpermissive]
 static libhelix::AACDecoderHelix _aac(aacAudioDataCallback);
                                                           ^
In file included from C:\Users\peter\Documents\Arduino\MiniTV\MJPEG_2task_Audio_1task\esp32_audio_task.h:3,
                 from C:\Users\peter\Documents\Arduino\MiniTV\MJPEG_2task_Audio_1task\MJPEG_2task_Audio_1task.ino:40:
C:\Users\peter\Documents\Arduino\libraries\arduino-libhelix-main\src/AACDecoderHelix.h:29:41: note:   initializing argument 1 of 'libhelix::AACDecoderHelix::AACDecoderHelix(libhelix::AACDataCallback)'
         AACDecoderHelix(AACDataCallback dataCallback){
                         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~
In file included from C:\Users\peter\Documents\Arduino\MiniTV\MJPEG_2task_Audio_1task\MJPEG_2task_Audio_1task.ino:40:
esp32_audio_task.h:111:59: error: invalid conversion from 'void (*)(MP3FrameInfo&, int16_t*, size_t)' {aka 'void (*)(_MP3FrameInfo&, short int*, unsigned int)'} to 'libhelix::MP3DataCallback' {aka 'void (*)(_MP3FrameInfo&, short int*, unsigned int, void*)'} [-fpermissive]
 static libhelix::MP3DecoderHelix _mp3(mp3AudioDataCallback);
                                                           ^
In file included from C:\Users\peter\Documents\Arduino\MiniTV\MJPEG_2task_Audio_1task\esp32_audio_task.h:4,
                 from C:\Users\peter\Documents\Arduino\MiniTV\MJPEG_2task_Audio_1task\MJPEG_2task_Audio_1task.ino:40:
C:\Users\peter\Documents\Arduino\libraries\arduino-libhelix-main\src/MP3DecoderHelix.h:38:41: note:   initializing argument 1 of 'libhelix::MP3DecoderHelix::MP3DecoderHelix(libhelix::MP3DataCallback, libhelix::MP3Type)'
         MP3DecoderHelix(MP3DataCallback dataCallback, MP3Type mp3Type=MP3Normal){
                         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~
Meerdere bibliotheken gevonden voor "WiFi.h"
Gebruikt: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\WiFi
Niet gebruikt: C:\Users\peter\Documents\Arduino\libraries\WiFiNINA
Niet gebruikt: C:\Program Files (x86)\Arduino\libraries\WiFi
Meerdere bibliotheken gevonden voor "SD.h"
Gebruikt: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\SD
Niet gebruikt: C:\Program Files (x86)\Arduino\libraries\SD
Bibliotheek WiFi op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\WiFi  wordt gebruikt
Bibliotheek FS op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\FS  wordt gebruikt
Bibliotheek LittleFS op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\LittleFS  wordt gebruikt
Bibliotheek SPIFFS op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\SPIFFS  wordt gebruikt
Bibliotheek FFat op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\FFat  wordt gebruikt
Bibliotheek SD op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\SD  wordt gebruikt
Bibliotheek SPI op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\SPI  wordt gebruikt
Bibliotheek SD_MMC op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\SD_MMC  wordt gebruikt
Bibliotheek Arduino_GFX-master op versie 1.3.6 in map: C:\Users\peter\Documents\Arduino\libraries\Arduino_GFX-master  wordt gebruikt
Bibliotheek Wire op versie 2.0.0 in map: C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.10\libraries\Wire  wordt gebruikt
Bibliotheek arduino-libhelix-main op versie 0.1 in map: C:\Users\peter\Documents\Arduino\libraries\arduino-libhelix-main  wordt gebruikt
Bibliotheek JPEGDEC op versie 1.2.8 in map: C:\Users\peter\Documents\Arduino\libraries\JPEGDEC  wordt gebruikt
exit status 1
invalid conversion from 'void (*)(AACFrameInfo&, int16_t*, size_t)' {aka 'void (*)(_AACFrameInfo&, short int*, unsigned int)'} to 'libhelix::AACDataCallback' {aka 'void (*)(_AACFrameInfo&, short int*, unsigned int, void*)'} [-fpermissive]


Do you have an idea how to solve that problem? 
Best regards
Peter

**Comment by animevietsub** [comment]
Make sure you use the old version of `arduino-libhelix` which is here:
[arduino-libhelix](https://github.com/jesserockz/arduino-libhelix)
or
[arduino-libhelix-drive](https://drive.google.com/file/d/1ZIDwWr6xl5vp7Q3TgRXrmRTwtJK29acn/view?usp=sharing)

**Comment by pvdw0310** [comment]
Hi, thanks for the reply.
I have no compiler errors.
The video plays perfect now, but I don't have any sound. 
If I check bclk the pin 25 of the esp32 there is no signal (no low or high or transition).     
If I check Dout the pin 33 of the esp32 it is always low. 
It looks like that there is no signal passing to the amplifier.    

I'am using an esp32 wemos d1 mini.

**Comment by animevietsub** [comment]
Can you send the serial console?

**Comment by pvdw0310** [comment]
I send you here the serial output as a picture.
![test](https://github.com/moononournation/MiniTV/assets/35037434/51625e36-b9fc-4571-8458-ec752662d9f7)

**Comment by animevietsub** [comment]
`OOM in SBR, can't allocate 50788 bytes` out of memory, make sure you have enough RAM to create SBR.
Try running other files like MP3.

**Comment by pvdw0310** [comment]
I changed the sound to mp3 and now it works. 
If the film is playing, I get now and then some black lines on screen. How to prevent this? 
Do I have to lower the frame rate?

**Comment by animevietsub** [comment]
Can you send the serial console and the image about the "some black lines on screen"?

**Comment by pvdw0310** [comment]
I have made a small video but it is to big to upload, so I took a frame out of the film. 
![fout](https://github.com/moononournation/MiniTV/assets/35037434/2c026e33-41eb-4b8d-a823-3213f84014fe)
For the moment everything is nicely in the 3d printed tv and I can't reach the reset button to view the startup on the arduino monitor, but at the end I get: 
![fout1](https://github.com/moononournation/MiniTV/assets/35037434/3fad0cbd-2214-478b-88d1-e75d7ee8b4c2)

I'am using the mp3 sound file now.

Everything is working, Only now and then some interrupted spots on the screen. It is not a hard thing, but If I can solve it, it would be better.
